### DISCUSSION TITLE
Environment variable when re-running actions

### DISCUSSION THREAD

**Discussion by kewisch** [community/community#10462] [Actions]
Hey folks. I'm wondering if it would be a good idea to have an environment variable set when a GitHub Action has been retriggered. The use case I'm imagining:

**Before**:

1) Workflow runs normally. Fails for some obscure reason.
2) Developer would like to debug, but then needs to enable debugging. Requires a commit to change.
3) Turns out it was an upstream or intermittent error, revert the commit.

**After**:

```yaml
  - name: "Random Step"
    uses: kewisch/action-random-example@v1
    with:
      debug: ${{ env.GITHUB_RUN_ATTEMPT > 1 }}
``` 

1) Workflow runs normally. Fails for some obscure reason.
2) Workflow re-run
3) Turns out it was an upstream or intermittent error, nothing to revert

Could also be used to skip certain steps if it is a re-run.

**Comment by chrispat** [comment]
Hello @kewisch thank you for your feedback.

We do provide a property that you can use to determine if the run is a rerun or not.  Looking at the[ `github` context](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) you will see a `run_attempt` property that indicates what attempt this is for the overall run.  

You can access this property in an expression `if: ${{ github.run_attempt > 1 }} ` or simply extract the value into an environment variable.
